pkg/solana/transmitter.go

```diff
@@ -92,16 +92,18 @@ func (c *ContractTracker) Transmit(
 
 	// Send transaction, and wait for confirmation:
 	go func() {
-		if _, err := confirm.SendAndConfirmTransactionWithOpts(
+		txSig, err := confirm.SendAndConfirmTransactionWithOpts(
 			context.Background(), // does not use libocr transmit context
 			c.client.rpc,
 			c.client.ws,
 			tx,
 			true, // skip preflight
 			rpc.CommitmentConfirmed,
-		); err != nil {
+		)
+		if err != nil {
```

Comment:
We should report this error back to libocr if it happens while libocr context still didn't expire (timeout).

For example, if the `SendAndConfirmTra..` errors immediately, we don't actually report these errors upstream, so libocr assumes success while more information is available.

Reply:
https://github.com/smartcontractkit/chainlink-solana/issues/58

Reply:
oooh that's a good point. I'll give it a shot...and see if i can test it too

Reply:
update: this will be done in a separate PR. this PR is now for getting everything working again with recent program changes